c/certificateService.c

```diff
@@ -118,36 +119,80 @@ static void respondWithBadRequest(HttpResponse *response) {
 
     jsonStart(p);
     {
-      jsonAddString(p, "error", "The length of the certificate is longer than 4096 bytes");
+      jsonAddString(p, "error", value);
     }
     jsonEnd(p);
 
     finishResponse(response);
 }
 
+
 static int serveMappingService(HttpService *service, HttpResponse *response)
 {
   HttpRequest *request = response->request;
 
   if (!strcmp(request->method, methodPOST))
   {
-    RUsermapParamList *userMapCertificateStructure 
-      = (RUsermapParamList*)safeMalloc31(sizeof(RUsermapParamList),"RUsermapParamList");
-    memset(userMapCertificateStructure, 0, sizeof(RUsermapParamList));
-
-    if(request->contentLength > sizeof(userMapCertificateStructure->certificate) || request->contentLength < 0) {
-      respondWithBadRequest(response);
-      return 0;
+    RUsermapParamList userMapStructure;
+
+    memset(&userMapStructure, 0, sizeof(RUsermapParamList));
+
+    char translatedURL[strlen(request->uri) +1];
+    memcpy(translatedURL, request->uri, strlen(request->uri));
+    a2e(translatedURL, sizeof(translatedURL) - 1);
+    char *found = strstr(translatedURL,"x509");
```

Comment:
`translatedURL` is not null-terminated. What could happen is,
* Someone sends an empty/incorrect request
* We still find `x509` further down the road, because someone else has used one before
* We use it for the wrong caller

Reply:
sorry, I think I made a mistake here, now I removed `-1`